test/libsolidity/semanticTests/abiEncoderV1/abi_decode_dynamic_array.sol

```diff
@@ -8,3 +8,7 @@ contract C {
 // compileViaYul: also
 // ----
 // f(bytes): 0x20, 0xc0, 0x20, 0x4, 0x3, 0x4, 0x5, 0x6 -> 0x20, 0x4, 0x3, 0x4, 0x5, 0x6
+// gas Legacy: 25792
+// gas LegacyOptimized: 23842
+// gas Yul: 26117
```

Comment:
In #10779 there seems to be a convergence to wards "legacy" and "ir" as the two options. Perhaps we should use those terms here too.

I am fine as is, and we can do another big round of renaming as part of the PR changing `compileViaYul`.

Reply:
Naming came up in https://github.com/ethereum/solidity/pull/10356#discussion_r565514043 too. Do you think we should use IR instead of Yul there?

Reply:
Will rename to "ir"